express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const data = require('./data.json');
+
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const json = express.json();
+
+app.use(json);
+
+app.get('/api/notes', (req, res) => {
+  const allNotes = [];
+  for (const keys in data.notes) {
+    allNotes.push(data.notes[keys]);
+  }
+  res.json(allNotes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const paramId = Number(req.params.id);
+  if (Number.isInteger(paramId) === false || Math.sign(paramId) !== 1) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (!data.notes[req.params.id]) {
+    res.status(404).send({ error: 'cannot find note with id ' + paramId });
+  } else {
+    res.json(data.notes[paramId]);
+  }
+});
+
+let nextId = 10;
+for (const keys in data.notes) {
+  const key = Number(keys);
+  if (key > nextId) {
+    nextId = key + 1;
+  }
+}
+app.post('/api/notes', (req, res) => {
+  if (Object.keys(req.body).length === 0) {
```

Comment:
What if there is a `"kiki"` property? That will be a key length of 1. It's much better to be explicit about what you are looking for. `typeof req.body.content === 'undefined'`